lib/core/src/Cardano/Wallet/Primitive/Types/Coin.hs

```diff
@@ -69,26 +76,21 @@ import Quiet
     ( Quiet (..) )
 
 import qualified Data.Text as T
+import qualified Prelude
 
 -- | A 'Coin' represents a quantity of lovelace.
 --
 -- Reminder: 1 ada = 1,000,000 lovelace.
 --
--- NOTE: The 'Coin' value is stored as a 64-bit unsigned integer. The maximum
--- supply of lovelace is less than 2^56, so there is ample space to store any
--- circulating amount of Ada.
---
--- However be careful when summing coins, for example, if calculating historical
--- volumes traded, because this may overflow.
 newtype Coin = Coin
-    { unCoin :: Word64
+    { unCoin :: Natural
     }
     deriving stock (Ord, Eq, Generic)
     deriving (Read, Show) via (Quiet Coin)
 
 instance Semigroup Coin where
```

Comment:
One of the most important things about the `Coin` abstraction is that it has a `Semigroup` and a `Monoid` instance which correspond to *summation*. Could you add corresponding comments in the Haddock documentation?

Reply:
Good point!

I've added some comments in 535d0dd85454b3d2ea1206c04ef13169c801c826.